connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -137,43 +136,49 @@ public void authenticate(String username, String password) {
             return;
         }
 
-        try {
-            MinecraftProtocol protocol;
-            if (password != null && !password.isEmpty()) {
-                protocol = new MinecraftProtocol(username, password);
-            } else {
-                protocol = new MinecraftProtocol(username);
-            }
-
-            downstream = new Client(remoteServer.getAddress(), remoteServer.getPort(), protocol, new TcpSessionFactory());
-            downstream.getSession().addListener(new SessionAdapter() {
-
-                @Override
-                public void connected(ConnectedEvent event) {
-                    loggedIn = true;
-                    connector.getLogger().info(authenticationData.getName() + " (logged in as: " + protocol.getProfile().getName() + ")" + " has connected to remote java server on address " + remoteServer.getAddress());
-                    playerEntity.setUuid(protocol.getProfile().getId());
-                    playerEntity.setUsername(protocol.getProfile().getName());
-                }
-
-                @Override
-                public void disconnected(DisconnectedEvent event) {
-                    loggedIn = false;
-                    connector.getLogger().info(authenticationData.getName() + " has disconnected from remote java server on address " + remoteServer.getAddress() + " because of " + event.getReason());
-                    upstream.disconnect(event.getReason());
+        loggedIn = true;
+        // new thread so clients don't timeout
+        new Thread(() -> {
```

Comment:
This should really be put onto a worker thread instead of creating and destroying threads.